md-dao/src/main/java/com/wipro/ats/bdre/md/dao/ProcessDAO.java

```diff
@@ -544,6 +544,84 @@ public Process cloneProcess(Integer processId) {
+                    preprocessingProcess = childProcess;
+                    preprocessingProcess.setNextProcessId(parentPid.toString());
+                    preprocessingProcess.setProcessId((Integer) session.save(preprocessingProcess));
+                }else  if (childProcess.getProcessType().getProcessTypeId() == 33){
+                    sourcestageloadProcess = childProcess;
+                    sourcestageloadProcess.setNextProcessId(parentPid.toString());
+                    sourcestageloadProcess.setProcessId((Integer) session.save(sourcestageloadProcess));
+                }
+                else  if (childProcess.getProcessType().getProcessTypeId() == 34){
+                    sourcetodeststagecopyProcess = childProcess;
+                    sourcetodeststagecopyProcess.setNextProcessId(parentPid.toString());
+                    sourcetodeststagecopyProcess.setProcessId((Integer) session.save(sourcetodeststagecopyProcess));
+                }
+                else  if (childProcess.getProcessType().getProcessTypeId() == 35){
+                    desttableloadProcess = childProcess;
+                    desttableloadProcess.setNextProcessId(parentPid.toString());
+                    desttableloadProcess.setProcessId((Integer) session.save(desttableloadProcess));
+                }
+                else  if (childProcess.getProcessType().getProcessTypeId() == 36){
+                    registerpartitionProcess = childProcess;
+                    registerpartitionProcess.setNextProcessId(parentPid.toString());
+                    registerpartitionProcess.setProcessId((Integer) session.save(registerpartitionProcess));
+                }
+
+                processList.add(childProcess);
+            }
+            parentProcess.setNextProcessId(preprocessingProcess.getProcessId().toString());
+            preprocessingProcess.setNextProcessId(sourcestageloadProcess.getProcessId().toString());
+            sourcestageloadProcess.setNextProcessId(sourcetodeststagecopyProcess.getProcessId().toString());
+            sourcetodeststagecopyProcess.setNextProcessId(desttableloadProcess.getProcessId().toString());
+            desttableloadProcess.setNextProcessId(registerpartitionProcess.getProcessId().toString());
+
+            session.update(parentProcess);
+            session.update(preprocessingProcess);
+            session.update(sourcestageloadProcess);
+            session.update(sourcetodeststagecopyProcess);
+            session.update(desttableloadProcess);
+
+            session.getTransaction().commit();
+        }
+        catch (MetadataException e) {
+            session.getTransaction().rollback();
+            LOGGER.error(e);
+        } finally {
+            session.close();
+        }
+        return processList;
+    }
     public List<Process> createDataloadJob(Process parentProcess, List<Process> childProcesses, List<Properties> parentProps, Map<Process,List<Properties>> childProps ){
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) The Cyclomatic Complexity of this method "createDataloadJob" is 16 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)